platform/jobs/edxPlatformQualityPr.groovy

```diff
@@ -0,0 +1,137 @@
+    String ghprbConfigContents = new File("${GHPRB_SECRET}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(secretFileContents)
+    ghprbMap = yaml.load(ghprbConfigContents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    assert jobConfig.containsKey('open')
+
+    job(jobConfig['jobName']) {
+
+        /* For non-open jobs, enable project based security */
+        if (!jobConfig['open'].toBoolean()) {
+            authorization {
+                blocksInheritance(true)
+                permissionAll('edx')
+            }
+        }
+        description('This job runs pull requests through our quality checks.')
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR()
+        properties {
+              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['platformUrl'])
+        }
+        concurrentBuild()
+        label('jenkins-worker')
+        scm {
+            git { //using git on the branch and url, clone, clean before checkout
+                remote {
+                    url(jobConfig['protocol'] + "://github.com/" + jobConfig['platformUrl'] + '.git')
+                    refspec('+refs/pull/*:refs/remotes/origin/pr/*')
+                    if (!jobConfig['open'].toBoolean()) {
+                        credentials(jobConfig['platformCredential'])
+                    }
+                }
+                branch('\${ghprbActualCommit}')
+                browser()
+                extensions {
+                    cloneOptions {
+                        reference("\$HOME/" + jobConfig['platformCloneReference'])
+                        timeout(10)
+                    }
+                    cleanBeforeCheckout()
+                    relativeTargetDirectory(jobConfig['repoName'])
+                }
+            }
+        }
+        triggers { //trigger when pull request is created
+           pullRequest {
+               admins(ghprbMap['admin'])
+                useGitHubHooks()
+                triggerPhrase('jenkins run quality')
+                // skip phrase
+                userWhitelist(ghprbMap['userWhiteList'])
+                orgWhitelist(ghprbMap['orgWhiteList'])
+               extensions {
+                  commitStatus {
+                        context('jenkins/quality')
+                    }
+                }
+             }
+        }
+        wrappers {
+            timeout {
+                absolute(45)
+            }
+            timestamps()
+            colorizeOutput()
+            buildName('#\${BUILD_NUMBER}: Quality Tests')
+        }
+        steps {
+            shell('cd edx-platform; TEST_SUITE=quality ./scripts/all-tests.sh')
```

Comment:
change from edx-platform to jobConfig['repoName'] to work with the other repo :)